### PR TITLE
IBP-3528-ValidateTrialInstanceValue

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Created logic for checking trial instance value 
Uses reusable util. validateVariableValues to check if trial instance has numeric value

### CODE REVIEW

File: src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java [hunk 1]
```diff
@@ -318,4 +325,24 @@ public FileUploadForm getForm() {
...
 	}
 
+	private void validateTrialInstanceValue(final List<MeasurementVariable> conditions, final List<MeasurementVariable> trialVariables, final Map<String, List<Message>> errorMessages) {
+		for (final MeasurementVariable varCondition : conditions) {
+			if (varCondition.getTermId() == 0) {
+				final List<MeasurementVariable> trialVariable = trialVariables.stream().filter(mVar -> {
+					return mVar.getScale().equals(varCondition.getScale()) && mVar.getMethod().equals(varCondition.getMethod()) && mVar.getProperty().equals(varCondition.getProperty());
+				}).collect(Collectors.toList());
+				if (trialVariable != null && !trialVariable.isEmpty()) {
+					varCondition.setTermId(trialVariable.get(0).getTermId());
```
Comment (abatac) [inline src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java:335]: Why do we copy the termId from trialVariables to condition?  I think getTrialVariables is just derived from Workbook.conditions?

Comment (gelli-leafnode) [reply src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java]: Hi @abatac , 
Logic added: set measurementid in condition variables

File: src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java [hunk 2]
```diff
@@ -318,4 +326,19 @@ public FileUploadForm getForm() {
 		return new FileUploadForm();
 	}
 
+	public void validateTrialInstanceValue(final List<MeasurementVariable> conditions, final Map<String, List<Message>> errorMessages) {
+		for (final MeasurementVariable varCondition : conditions) {
+			if (varCondition.getTermId() == TermId.TRIAL_INSTANCE_FACTOR.getId()) {
+				final Optional<Message> message = Util.validateVariableValues(varCondition, varCondition.getValue());
+				if (message.isPresent()) {
+					if (errorMessages.containsKey(Constants.INVALID_TRIAL)) {
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java:334]: if-else can be simplified as:
'errorMessages.putIfAbsent(Constants.INVALID_TRIAL, new ArrayList<>());
					errorMessages.get(Constants.INVALID_TRIAL).add(message.get());'

File: src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java [hunk 3]
```diff
@@ -190,6 +193,11 @@ private int getAllocatedCount(final Map<PhenotypicType, List<VariableDTO>> heade
 				proxy.put("", this.etlService.convertMessageList(messageList));
 			}
 
+			// Validate Trial Instance Value
+			if (importData.getConditions() != null && importData.getTrialVariables() != null) {
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java:197]: Use `!CollectionUtils.isEmpty(importData.getConditions())`

Comment (darla-leafnode) [inline src/main/java/com/efficio/etl/web/controller/angular/AngularMapOntologyController.java:197]: Why do we need this checking: `importData.getTrialVariables() != null`

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**